octodns/provider/ns1.py

```diff
@@ -193,22 +197,50 @@ def notifylists_list(self):
         return self._try(self._notifylists.list)
 
     def records_create(self, zone, domain, _type, **params):
-        return self._try(self._records.create, zone, domain, _type, **params)
+        cached = self._records_cache.setdefault(zone, {}) \
+            .setdefault(domain, {})
+        cached[_type] = self._try(self._records.create, zone, domain, _type,
+                                  **params)
+        return cached[_type]
 
     def records_delete(self, zone, domain, _type):
+        try:
+            # remove record from cache
+            del self._records_cache[zone][domain][_type]
+            # remove record's zone from cache
+            del self._zones_cache[zone]
```

Comment:
@viranch any specific reasons behind clearing the zone when records are deleted or updated?

Reply:
Ah. I'd forgotten that basic record info was in the zone and `records_retrieve` is only called when the record is `tier > 1`. 

I think that means we need to make sure to "clear" the zone when deleting it even when there's nothing in `records_cache` since most things won't be cached there (only `dynamic` stuff.)

Reply:
Yeah zones contain the records also. Good catch clearing zone cache first.